cadc-access-control-identity/src/main/java/ca/nrc/cadc/ac/ACIdentityManager.java

```diff
@@ -229,44 +242,58 @@ public Object toOwner(Subject subject) {
             if (principal instanceof X500Principal) {
                 x500Principal = (X500Principal) principal;
             }
+            if (principal instanceof OpenIdPrincipal) {
+                openIdPrincipal = (OpenIdPrincipal) principal;
+            }
         }
 
-        if (x500Principal == null) {
+        NumericPrincipal numericPrincipal;
```

Comment:
In principal, a user could authenticate with an unknown OpenID *and* client cert... createAuthUser could make sure both are included in the new account, eg change to `createAuthUser(subject)`. Currently the code is chosing one type over the other.

Reply:
Not quite: at this point both the OpenID and X509 identifies are _new_ so it is not linking; it is someone providing two identities in one go.

If someone authenticated with both a token and cert and the OpenID was _new_ but the cert was not we would ignore the OpenID (there would be a NumericPrincipal). Likewise, if the cert was _new_ but the OpenID was not we would ignore the cert. Is that OK or bad? But _ignore_ here means it might still be in the subject, just not linked/interchangeable with other identifies in the same subject...

Reply:
I can see problems down the road if we allow bot cert and token. The second authentication might be used downstream and could trigger a different automatic account creation. Maybe I should check for it everywhere and fail it until we figure out a way to deal with this case.

Reply:
I think accepting it fully or failing are both better than silently chosing one ident and ignoring the other, even though this is a very obscure usage. I'd be OK with fail for now.

Reply:
Done